sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -777,17 +768,65 @@ object GpuCast extends Arm {
     }
   }
 
+  def castStringToDecimal(
+      input: ColumnView,
+      ansiEnabled: Boolean,
+      dt: DecimalType): ColumnVector = {
+    // 1. Sanitize strings to make sure all are floating points
+    // 2. Identify all fixed point values
+    // 3. Cast String to newDt (newDt = dt.scale + 1). Promote precision if needed. This step is
+    //    required so we can round up if needed in the final step
+    // 4. Now cast to newDt to dt (Decimal to Decimal)
+    def getInterimDecimalPromoteIfNeeded(dt: DecimalType): DecimalType = {
+      if (dt.scale + 1 > dt.precision) {
+        // promote if possible or throw
+        if (dt.precision == Decimal.MAX_LONG_DIGITS) {
+          //We don't support Decimal 128
+          throw new IllegalArgumentException("One or more values exceed the maximum supported " +
+              "Decimal precision while conversion")
+        }
+        DecimalType(dt.precision + 1, dt.scale + 1)
+      }
+      DecimalType(dt.precision, dt.scale + 1)
+    }
+
+    withResource(input.strip()) { trimmed =>
+      withResource(GpuCast.sanitizeStringToFloat(trimmed, ansiEnabled, false)) { sanitized =>
```

Comment:
`sanitizeStringToFloat` feels like a lot of overkill in this situation.  Do we really need that full regexp? Will `isFixedPoint` cover us? If not fully we can at least make the regexp a lot simpler, which should make it faster to process.  Also technically I don't think we support the [fFdD] for decimal types?  We should test that though.